stat/monitor.go

```diff
@@ -0,0 +1,81 @@
+package stat
+
+import (
+	"sync"
+	"time"
+)
+
+// methodStatus provide statistic for specific method.
+type methodStatus struct {
+	name string
+	mu   sync.RWMutex // protect counters
+	Snapshot
+}
+
+func (m *methodStatus) snapshot() Snapshot {
+	m.mu.RLock()
+	defer m.mu.RUnlock()
+	return m.Snapshot
+}
+
+func (m *methodStatus) incRequests(elapsed time.Duration) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	m.allTime += uint64(elapsed)
+	m.allRequests++
+}
+
+// Snapshot represents statistic for specific method.
+type Snapshot struct {
+	allTime     uint64
+	allRequests uint64
+}
+
+// AllTime returns sum of time, spent to specific request. Use with [time.Duration] to get human-readable value.
+func (s Snapshot) AllTime() uint64 {
+	return s.allTime
+}
+
+// AllRequests returns amount of requests to node.
+func (s Snapshot) AllRequests() uint64 {
+	return s.allRequests
+}
+
+type nodeMonitor struct {
+	pubKey         []byte
+	addr           string
+	errorThreshold uint32
+	methods        []*methodStatus
+
+	mu                sync.RWMutex // protect counters
+	overallErrorCount uint64
+}
+
+func (c *nodeMonitor) incErrorRate() {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.overallErrorCount++
+}
+
+func (c *nodeMonitor) overallErrorRate() uint64 {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+	return c.overallErrorCount
+}
+
+func (c *nodeMonitor) methodsStatus() []Snapshot {
+	result := make([]Snapshot, len(c.methods))
```

Comment:
can you explain a little why you decided to return it without names? relying on method order isn't error-prone? also, you still _do have_ names in some structs and do not use them

Reply:
It is just a copy of pool's statistic, I've tried to left it close as possible to the previous one. I don't mind to discuss and change format and other, but in the separate task

Reply:
> but in the separate task

ok

> It is just a copy of pool's statistic

oh, well, that is what i said about: if a commit message said to me that it is only a copy without any changes i would not review it

Reply:
Ok I see one more good reason to extend commit messages, I'm will work on myself